### PR TITLE
Provide instance set-up instructions

### PR DESCRIPTION
**Pull request by cpcallen**
Transcribe previous "bare metal" Google Doc into markdown.  Also:

* Use `codecity` username instead of `cc`.
* Fix issue `#!` parsing on Linux.

### CODE REVIEW

**Comment by cpcallen** [comment]
Ping.

File: docs/setup.md [hunk 1]
```diff
@@ -0,0 +1,702 @@
...
+In order to allow incoming HTTPS connections, you will need an
+[TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security) [server
+certificate](
+https://en.wikipedia.org/wiki/Public_key_certificate#TLS/SSL_server_certificate
+).  There are two types:
+
+*   An ordinary certificate covers one or more specific domain names,
+    like `www.example.org`.
+*   A wildcard certificate includes one or more wildcard domains, like
+    `*.example.org`.  If you’re using a wildcard DNS entry, a
```
Comment (NeilFraser) [inline docs/setup.md:264]: a...

File: docs/setup.md [hunk 2]
```diff
@@ -0,0 +1,702 @@
...
+There are various ways to get a TLS certificate, but a free and easy
+way is to use [Certbot](https://certbot.eff.org/) to get one from
+[Let’s Encrypt](https://letsencrypt.org/).  That’s what we’ll do here.
+
+#### Getting a wildcard certificate
+
+To use Certbot to get a wildcard certificate, you will need to use the
+[`dns-01` challenge](
+https://letsencrypt.org/docs/challenge-types/#dns-01-challenge), which
+requires being able to create [DNS TXT records](
```
Comment (NeilFraser) [inline docs/setup.md:280]: Can I recommend using [callout-links](https://www.google.com/search?q=external+link+icon) for what are essentially [dictionary](https://www.oed.com/) definitions.  This reserves actual [links](https://www.directindustry.com/industrial-manufacturer/lifting-chain-61038.html) for things that [people](https://i.redd.it/fxq431q8hqe51.jpg) are expected to follow.

File: docs/setup.md [hunk 3]
```diff
@@ -0,0 +1,702 @@
...
+    call the account `codecity` here, but any username is fine:
+    ```
+    sudo useradd -rms /bin/bash codecity
+    ```
+0.  Become the code city account:
+    ```
+    sudo -iu codecity
+    ```
+0.  Clone the [Code City
+    repo](https://github.com/NeilFraser/CodeCity).  (If you are a
```
Comment (NeilFraser) [inline docs/setup.md:383]: Now a Google project.

File: docs/setup.md [hunk 4]
```diff
@@ -0,0 +1,702 @@
...
+    project collaborator, see below for [instructions on how to use
+    SSH instad of HTTPS](#git-code-city-by-ssh-instead-of-https).)
+    ```
+    git clone https://github.com/google/CodeCity.git
+    ```
+0.  Install install required NPMs:
```
Comment (NeilFraser) [inline docs/setup.md:389]: Let's 'install' it a few more times to be sure.

File: docs/setup.md [hunk 5]
```diff
@@ -0,0 +1,702 @@
...
+TLS certificate because it depends on `logo-auth.png` being served by
+nginx.
+
+1.  Optionally replace `~codecity/CodeCity/static/logo-auth.png` with
+    a logo representing your instance or organisation.  It should be a
+    120x120px PNG image.
+0.  Make sure you can access your desired logo using your web browser.
+    If you are using a wildcard DNS configuration, it should be
+    accessible via a URL like
+    <code>static.<em>exmaple</em>.codecity.world/logo-auth.png</code>;
```
Comment (NeilFraser) [inline docs/setup.md:453]: 'exmaple'
And two lines down.

File: docs/setup.md (hunk 1 above)
Comment (cpcallen) [reply docs/setup.md:264]: Actually, the whole "If you’re using a wildcard DNS entry, a" is surplus to requirements.

File: docs/setup.md (hunk 2 above)
Comment (cpcallen) [reply docs/setup.md:280]: This seems reasonable, but I am entirely unclear about what a callout link is, or how I go about creating one in markdown.

Can you give a concrete example?

File: docs/setup.md (hunk 5 above)
Comment (cpcallen) [reply docs/setup.md:453]: No, this is correct; I was referring to the code city instance run by former employees of the well known Waterloo software firm [Maple](https://www.maplesoft.com/).  ;-)

**Comment by cpcallen** [comment]
I've gone ahead and merged this but would be happy to make further update with callout links once I understand what you have in mind.

**Comment by NeilFraser** [comment]
Superscript is one way to do a link <sup>[[?]](https://neil.fraser.name)</sup>
Or another way would is to use an image [](https://neil.fraser.name)

**Comment by cpcallen** [comment]
Oh, you are proposing I format 'dictionary definition' links this way?  I can do that.

**Pull request merged**